### PR TITLE
Content collections refactor

### PR DESCRIPTION
**Pull request by bholmesdev**
- Bump dependencies to beta for Astro core, `@astrojs/image`, `@astrojs/mdx`, and `@astrojs/tailwind`
- Migrate blog post parser to a collection schema
- Refactor `getAuthor(id)` to better use schemas
- Update `BaseHead` and `BlogPost` templates to use content collections
- Replace explicit `zod` dependency with imports from `astro/zod`

### CODE REVIEW

File: src/components/Mention.astro [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
+import { getAuthor } from '../data/authors'
+import Avatar from './Avatar.astro'
+const { name } = Astro.props
 
-const mention = await getAuthor(name);
-if (!mention) {
-    throw new Error(`Could not find mentioned user "${name}"!`)
-}
+const mention = getAuthor(name)
+if (mention instanceof Error) throw Error
```
Comment (itsMapleLeaf) [inline src/components/Mention.astro:7]: looks like this should be `throw mention`

Although, this looks the same as `getAuthor` throwing the error itself, but with extra steps :thinking: `getAuthor` doesn't even appear to return `Error`

Comment (bholmesdev) [reply src/components/Mention.astro:7]: Ah yep sorry, used to return the error object, and it now throws internally. Good to remove this

**Comment by tony-sull** [comment]
@bholmesdev Netlify is pulling the Node version from `.nvmrc`.  For the 2.0 beta are smoke tests using Node 16 now?

File: src/pages/blog/_utils.ts [hunk 2]
```diff
@@ -1,48 +1,8 @@
-import path from 'node:path';
```
Comment (tony-sull) [inline src/pages/blog/_utils.ts:1]: bye bye, boilerplate code 🚀

**Comment by bholmesdev** [comment]
TODO: bump to non-beta versions and merge!

**Review by tony-sull — APPROVED** [review]
lgtm 💰

**Pull request merged**